### DISCUSSION TITLE
JSON db on Vercel like lowdb?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [vercel/next.js#17949] [Help]
Is possible to run a small JSON db on Vercel like lowdb? (only for playing around). Thanks

**Comment by dinaramaro** [comment]
Hello,
I've just created an Express server with lowdb in a Next.js project, it works well!

**Reply by krystiansliwa** [reply]
Could you share more details? ;)

**Comment by dinaramaro** [comment]
I have created my Express server in my project Next.js (file ssr-server.js
at the top level of my project) and imported lowdb in it.
To create the Express server, you need to import also next and call the
function getRequestHandler() const handle = app.getRequestHandler();
Then create your const app :
const dev = process.env.NODE_ENV !== 'production';
const app = next({ dev });
Call the prepare() function on app and initiate Express.
At the end you use a wildcard route to catch all routes and return it to
the handler function  server.get('*', (req, res) => { return handle(req, res
)});
Here is the tuto I followed to create my Express server :
https://blog.logrocket.com/how-to-build-a-server-rendered-react-app-with-next-express-d5a389e7ab2f/

const express = require('express')const next = require('next')
    const dev = process.env.NODE_ENV !== 'production'const app =
next({ dev })const handle = app.getRequestHandler()

app.prepare().then(() => {
  const server = express()

  server.get('*', (req, res) => {
    return handle(req, res)
  })

  server.listen(3000, (err) => {
    if (err) throw err
    console.log('> Ready on http://localhost:3000')
  })}).catch((ex) => {
  console.error(ex.stack)
  process.exit(1)})


Le jeu. 29 oct. 2020 à 10:29, Krystian Śliwa <notifications@github.com> a
écrit :

> Could you share more details? ;)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/next.js/discussions/17949#discussioncomment-115356>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEMMSM7UXWKGMAI3YJEHILSNEYWJANCNFSM4STSK5RQ>
> .
>


-- 
Dina Ramarovahoaka.

<https://www.facebook.com/dina.ramarovahoaka>
<https://www.linkedin.com/in/dina-ramarovahoaka-97858077/>
<https://twitter.com/dina_ramaro> <https://www.instagram.com/dinaramaro/>
<https://fr.pinterest.com/dinaramaro/>

**Reply by krystiansliwa** [reply]
Thank you @dinaramaro I will check it in free time :)

**Comment by dinaramaro** [comment]
You're welcome :)

Le jeu. 29 oct. 2020 à 11:23, Krystian Śliwa <notifications@github.com> a
écrit :

> Thank you @dinaramaro <https://github.com/dinaramaro> I will check it in
> free time :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/next.js/discussions/17949#discussioncomment-115407>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEMMSIOYKACITKMBD3AJMTSNE7D5ANCNFSM4STSK5RQ>
> .
>


-- 
Dina Ramarovahoaka.

<https://www.facebook.com/dina.ramarovahoaka>
<https://www.linkedin.com/in/dina-ramarovahoaka-97858077/>
<https://twitter.com/dina_ramaro> <https://www.instagram.com/dinaramaro/>
<https://fr.pinterest.com/dinaramaro/>
